src/modalities/models/gpt2/huggingface_model.py

```diff
@@ -1,25 +1,25 @@
 from typing import Dict
 
 import torch
+import torch.nn as nn
 from transformers import PreTrainedModel
 
-from modalities.config.config import PretrainedGPTConfig
+from modalities.config.config import HuggingFaceModelConfig
 from modalities.models.gpt2.gpt2_model import GPT2LLM
 
 
-class PretrainedGPTModel(PreTrainedModel):
-    config_class = PretrainedGPTConfig
+class HuggingFaceModel(PreTrainedModel):
+    config_class = HuggingFaceModelConfig
 
-    def __init__(self, config: PretrainedGPTConfig):
+    def __init__(self, config: HuggingFaceModelConfig, model: nn.Module = None):
         super().__init__(config)
         # TODO offloading the parameters like this is ugly
-        self.model: GPT2LLM = GPT2LLM(**dict(config.config))
+        if model is None:
+            self.model: GPT2LLM = GPT2LLM(**dict(config.config))
```

Comment:
Same as for the config: The forward- and prepare_inputs_for_generation-functions may differ for different model architectures. Maybe we rename HuggingFaceModel to GPT2HuggingFaceModel?

Reply:
@le1nux friendly reminder...

Reply:
This class and ModalitiesModelOutput should be at lest renamed as Richard proposed.
Ideally, there would be a way to at least have an abstract class that partially provides the Huggingface-Wrapping functionalities for our models. But this might be hard, especially since the parameters of forward() might change depending on the modalities of other models.

Reply:
We created an issue for that: 
https://github.com/Modalities/modalities/issues/71